stdlib/public/runtime/Casting.cpp

```diff
@@ -169,7 +169,9 @@ void
 swift::swift_dynamicCastFailure(const void *sourceType, const char *sourceName, 
                                 const void *targetType, const char *targetName, 
                                 const char *message) {
+#if !defined(_WIN32)
   asm("");
```

Comment:
(If the function is empty and you use noinline without asm("") then the optimizer may simply delete the call at the call site. Technically that doesn't count as inlining. asm("") inhibits that optimization.)